### DISCUSSION TITLE
How to highlight some words in a verbatim block

### DISCUSSION THREAD

**Discussion by jandermoreira** [quarto-dev/quarto-cli#7697] [Q&A] [answered]
### Description

Is there a way to highlight specific words in a code block?

For example, consider this block to simulate an interactive program execution:

````
```
What is your name? John Doe
Hello, John Doe!
```
````

I wish what the user typed to have a distinct color or be highlighted someway.

Something like:

What is your name? $\mbox{\color{blue}\large John Doe}$
Hello, John Doe!

**Comment by mcanouil** [comment]
I don't think you can or you would need to define a highlight style as for programming languages.
Or don't use markdown.

**Comment by yoonghm** [comment]
I think you could create your own syntax. See https://hamel.dev/notes/quarto/highlighting.html

**Reply by mcanouil** [reply]
Indeed, I was referring to this Pandoc/Quarto ability to use highlighting.
Thank you!

**Comment by jandermoreira** [comment, marked as answer]
I ended writing my own simple language, thank you.

[terminal.xml.zip](https://github.com/quarto-dev/quarto-cli/files/13539183/terminal.xml.zip)

It just highlights text between null characters to differentiate user input from the rest. I opted to use `\0` to render "invisible" characters. Since my examples are generated by  a program, I don't have to type de null characters. 

It works with HTML and I had to add `\catcode0=9` to allow nulls in TeX files.

To work with Quarto:

```
syntax-definitions: 
  - terminal.xml
```
